core/logger/zap.go

```diff
@@ -5,27 +5,89 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"path/filepath"
 
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
 )
 
 var _ Logger = &zapLogger{}
 
+type ZapLoggerConfig struct {
+	zap.Config
+	local *Config
```

Comment:
Maybe embed `local` so you don't have to worry about `nil` anywhere below?
```suggestion
	local Config
```